### PR TITLE
Fix actual bug

### PR DESCRIPTION
**Pull request by AndrewJWin**

### CODE REVIEW

File: commands/fun/say.js [hunk 1]
```diff
@@ -13,24 +13,29 @@ class Say extends Social {
...
-    if (!channel || msg.length < 1) throw new this.client.methods.errors.UsageError("You need to give the bot a message to send.", loadingMessage);
-    msg = msg.join(" ");
+  async cmdVerify(message, args, loadingMessage) {
+    if (args.length < 1) throw new this.client.methods.errors.UsageError("You need to give the bot a message to send.", loadingMessage);
+    const channelid = await this.verifyChannel(message, args[0], { msg: loadingMessage });
+    if (channelid !== message.channel.id) {
+      args.shift();
+    }
+    const channel = message.guild.channels.get(channelid);
+    if (channel.permissionsFor(message.member).missing(["SEND_MESSAGES", "VIEW_CHANNEL"]).length > 0) throw new this.client.methods.errors.UsageError("You do not have permission to `say` in that channel.", loadingMessage);
```
Comment (MrJacz) [inline commands/fun/say.js:24]: use `.has`

File: commands/fun/say.js [hunk 2]
```diff
@@ -13,24 +13,29 @@ class Say extends Social {
...
+    if (channel.permissionsFor(message.guild.me).missing(["SEND_MESSAGES", "VIEW_CHANNEL"]).length > 0) throw new this.client.methods.errors.UsageError("I do not have permission to `say` in that channel.", loadingMessage);
```
Comment (MrJacz) [inline commands/fun/say.js:25]: same here

File: commands/fun/say.js [hunk 3]
```diff
@@ -13,24 +13,29 @@ class Say extends Social {
...
   }
 
-  async run(message, [channel, ...msg], level, loadingMessage) {
 
-    if (!channel || msg.length < 1) throw new this.client.methods.errors.UsageError("You need to give the bot a message to send.", loadingMessage);
-    msg = msg.join(" ");
+  async cmdVerify(message, args, loadingMessage) {
+    if (args.length < 1) throw new this.client.methods.errors.UsageError("You need to give the bot a message to send.", loadingMessage);
+    const channelid = await this.verifyChannel(message, args[0], { msg: loadingMessage });
+    if (channelid !== message.channel.id) {
```
Comment (MrJacz) [inline commands/fun/say.js:20]: this could be shorten to one line

File: commands/fun/say.js [hunk 4]
```diff
@@ -13,24 +13,29 @@ class Say extends Social {
...
+  async run(message, args, level, loadingMessage) {
+    const channel = await this.cmdVerify(message, args, loadingMessage);
 
-    const channelid = await this.verifyChannel(message, channel, { msg: loadingMessage });
-    if (channelid === message.channel.id) msg = channel + " " + msg;    
-    const chn = message.guild.channels.get(channelid);
-    if (chn.permissionsFor(message.guild.me).missing(["SEND_MESSAGES", "READ_MESSAGES"]).length > 0) throw new this.client.methods.errors.UsageError("I do not have permission to `say` in that channel.", loadingMessage);
-    if (chn.permissionsFor(message.member).missing(["SEND_MESSAGES", "READ_MESSAGES"]).length > 0) throw new this.client.methods.errors.UsageError("You do not have permission to `say` in that channel.", loadingMessage);
-    
     message.delete();
```
Comment (MrJacz) [inline commands/fun/say.js:32]: Missing await

**Review by MrJacz — CHANGES_REQUESTED** [review]

File: commands/fun/say.js (hunk 1 above)
Comment (AndrewJWin) [reply commands/fun/say.js:24]: It

File: commands/fun/say.js (hunk 2 above)
Comment (AndrewJWin) [reply commands/fun/say.js:25]: is

File: commands/fun/say.js (hunk 3 above)
Comment (AndrewJWin) [reply commands/fun/say.js:20]: called

File: commands/fun/say.js (hunk 4 above)
Comment (AndrewJWin) [reply commands/fun/say.js:32]: Yorks Code

**Pull request closed without merging**